src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs

```diff
@@ -44,32 +58,35 @@ public AuthorizationLevelAttribute(AuthorizationLevel level)
                 // as a request property
                 var secretManager = actionContext.ControllerContext.Configuration.DependencyResolver.GetService<ISecretManager>();
 
-                requestAuthorizationLevel = await GetAuthorizationLevelAsync(request, secretManager, EvaluateKeyMatch);
-                request.SetAuthorizationLevel(requestAuthorizationLevel);
+                var result = await GetAuthorizationResultAsync(request, secretManager, EvaluateKeyMatch, KeyName);
+                requestAuthorizationLevel = result.AuthorizationLevel;
```

Comment:
I don't think we need this assignment 
`requestAuthorizationLevel` is not being used in the rest of the function body

in the old code, it is used to compare against Level:
`if (requestAuthorizationLevel < Level)`
but in the new code 
`if (!request.HasAuthorizationLevel(Level))`
is extracting requestAuthorizationLevel itself